lib/stripe/stripe_client.rb

```diff
@@ -657,17 +658,24 @@ def self.maybe_gc_connection_managers
     end
 
     private def notify_request_end(context, duration, http_status, num_retries,
-                                   user_data)
+                                   user_data, resp, headers)
       return if !Instrumentation.any_subscribers?(:request_end) &&
                 !Instrumentation.any_subscribers?(:request)
 
-      event = Instrumentation::RequestEndEvent.new(
+      request_context = Stripe::Instrumentation::RequestContext.new(
         duration: duration,
+        context: context,
+        header: headers
+      )
+      response_context = Stripe::Instrumentation::ResponseContext.new(
         http_status: http_status,
-        method: context.method,
+        response: resp
+      )
+
```

Comment:
I wanted to instantiate both `request_context` and `response_context` inside of the `execute_request_with_rescues` method and just pass them into this `notify_request_end` method instead, but Rubocup lint was complaining that `execute_request_with_rescues` method had too many lines 🙈